tests/integration/test_dbapi_integration.py

```diff
@@ -859,3 +860,78 @@ def test_info_uri(trino_connection):
     cur.fetchall()
     assert cur.info_uri is not None
     assert cur._query.query_id in cur.info_uri
+
+
+def test_client_tags_single_tag(run_trino):
+    _, host, port = run_trino
+
+    client_tags = ["foo"]
+    trino_connection = trino.dbapi.Connection(
+        host=host,
+        port=port,
+        user="test",
+        source="test",
+        max_attempts=1,
+        client_tags=client_tags,
+    )
+    cur = trino_connection.cursor()
+    cur.execute('SELECT 1')
+    cur.fetchall()
+
+    api_url = "http://" + trino_connection.host + ":" + str(trino_connection.port)
+    query_info = requests.post(api_url + "/ui/login", data={
+        "username": "admin",
+        "password": "",
+        "redirectPath": api_url + '/ui/api/query/' + cur._query.query_id
+    }).json()
+    assert query_info['session']['clientTags'] == client_tags
+
+
+def test_client_tags_multiple_tags(run_trino):
+    _, host, port = run_trino
+
+    client_tags = ["foo", "bar"]
+    trino_connection = trino.dbapi.Connection(
+        host=host,
+        port=port,
+        user="test",
+        source="test",
+        max_attempts=1,
+        client_tags=client_tags,
+    )
+    cur = trino_connection.cursor()
+    cur.execute('SELECT 1')
+    cur.fetchall()
+
+    api_url = "http://" + trino_connection.host + ":" + str(trino_connection.port)
+    query_info = requests.post(api_url + "/ui/login", data={
+        "username": "admin",
+        "password": "",
+        "redirectPath": api_url + '/ui/api/query/' + cur._query.query_id
+    }).json()
+    assert query_info['session']['clientTags'] == client_tags
+
+
+def test_client_tags_special_characters(run_trino):
+    _, host, port = run_trino
+
+    client_tags = ["foo %20", "bar=test"]
+    trino_connection = trino.dbapi.Connection(
+        host=host,
+        port=port,
+        user="test",
+        source="test",
+        max_attempts=1,
+        client_tags=client_tags,
+    )
+    cur = trino_connection.cursor()
+    cur.execute('SELECT 1')
+    cur.fetchall()
+
+    api_url = "http://" + trino_connection.host + ":" + str(trino_connection.port)
+    query_info = requests.post(api_url + "/ui/login", data={
+        "username": "admin",
+        "password": "",
+        "redirectPath": api_url + '/ui/api/query/' + cur._query.query_id
+    }).json()
```

Comment:
Could you add a helper method that takes a `client_tags` argument and returns `query_info['session']['clientTags']`?

Reply:
Let's move `trino.dbapi.Connection` to this method.